### ISSUE TITLE
impl `JackHandler` for `FnMut + Send + 'static`

### ISSUE THREAD

**Issue by cramertj** [wmedrano/rust-jack#21]
Most `JackHandler`s are probably just going to be an implementation of the `process` method. It would be convenient to allow users to pass a closure as a `JackHandler`.

**Comment by wmedrano** [comment]
Took a stab at it #26
Why does it need the static?
If my model of lifetimes is correct, then the port's would also need to be static in this case.

**Comment by cramertj** [comment]
Yes, the ports should also need to be static in order to ensure soundness.

It's the whole "`mem::forget` is stable" thing. If you call `mem::forget` on a `Client`, the client's registered callbacks will continue to be triggered for the duration of the process (unless you manually deregister them later on). Therefore, any resources used in the callback must be valid for the `'static` lifetime.

**Comment by cramertj** [comment]
To give an example, here's a modified version of the `playback_capture` example that demonstrates this unsoundness:

```rust
 let x = 5;
 let x_ref = &x;
 let process_callback =
         ProcessClosure::new(move |ps: &jack::ProcessScope| -> jack::JackControl {
             // x_ref must still be valid here!
             let closure_x = *x_ref;
             out_a.data(ps).buffer().clone_from_slice(in_a.data(ps).buffer());
             out_b.data(ps).buffer().clone_from_slice(in_b.data(ps).buffer());
             jack::JackControl::Continue
         });
 // Activate the client, which starts the processing
 let active_client = client.activate(process_callback).unwrap();
 mem::forget(active_client);
 // Client is forgotten but not deregistered, stack pops up and x is lost
 // Callback then dereferences x_ref and gains access to stack garbage
```

**Comment by cramertj** [comment]
Fixed in #26
